### PR TITLE
CSV syntax support

### LINKED ISSUE #267: Validator: handleCsv() works incorrect
selection_001

```ls
csv rows from https://raw.githubusercontent.com/axibase/atsd-use-cases/master/research/us-visa-refusal/resources/visa-refusal.csv
        
      for row in rows
        [series]
          replace-value = value - @{row[2006]}
          [tags]
            country = @{row.country.replace(",", "\\,")}
      endfor
```

Also fix Formatter. Working config [here](https://apps.axibase.com/chartlab/b00d77c0).

### PR DESCRIPTION
**Pull request by PolinaShneider**
Closes #267 

Doesn't crush on syntax `csv <name> from <url>` anymore
Doesn't fetch data from url yet

### CODE REVIEW

File: server/src/validator.ts [hunk 1]
```diff
@@ -697,20 +697,35 @@ export class Validator {
...
+            let match: RegExpExecArray | null;
+
+            if (equals.test(line)) {
+                match = equals.exec(line);
+                header = line.substring(match.index + 1);
+            } else if (from.test(line)) {
+                match = from.exec(line);
+                header = line.substring(match.index + match[0].length);
+            } else {
+                throw new Error("Invalid CSV expression");
```
Comment (annstriganova) [inline server/src/validator.ts]: This error will appear in logs, it's not useful. Let's add diagnostic with level **Error** to inform user, use `handleElse()` as example.

File: server/src/validator.ts [hunk 2]
```diff
@@ -697,20 +697,35 @@ export class Validator {
...
-            if (match === null) {
-                throw new Error("The line does not contain a '='");
+            let match: RegExpExecArray | null;
+
+            if (equals.test(line)) {
+                match = equals.exec(line);
+                header = line.substring(match.index + 1);
+            } else if (from.test(line)) {
+                match = from.exec(line);
+                header = line.substring(match.index + match[0].length);
```
Comment (annstriganova) [inline server/src/validator.ts]: Since there is no access to the whole CSV, no need to count header columns.

File: server/src/validator.ts [hunk 3]
```diff
@@ -697,20 +697,35 @@ export class Validator {
...
+            } else {
+                throw new Error("Invalid CSV expression");
             }
-            header = line.substring(match.index + 1);
         }
-        this.match = /(^[ \t]*csv[ \t]+)(\w+)[ \t]*=/m.exec(line);
+        this.match = /(^[ \t]*csv[ \t]+)(\w+)[ \t]*(=|from)/m.exec(line);
         this.addToStringMap(this.variables, "csvNames");
         this.csvColumns = (header === null) ? 0 : countCsvColumns(header);
     }
```
Comment (VeselovAlex) [inline server/src/validator.ts]: Do we validate usage of csv columns in placeholders? 
```
csv servers =
  name, cpu_load
  s1, 23
  s2, 13
endcsv
for srv in servers
[series]
  label = @{srv.name}
  value = @{srv.cpuload} # should be an error
endfor
```

What happens for remote csv files (`csv ... from ...`) in this case

Comment (PolinaShneider) [reply server/src/validator.ts]: I have checked csv column names aren't validated. Will create an issue for this

File: server/src/test/csv.test.ts [hunk 4]
```diff
@@ -99,7 +99,40 @@ endcsv`,
...
+            "Incorrect csv 'from' with missing entity name",
+            `csv from https://raw.githubusercontent.com/axibase/atsd-use-cases/master/USVisaRefusal/Resources/visa-refusal.csv`,
+            [
+                createDiagnostic(
+                    Range.create(Position.create(0, 0), Position.create(0, "csv".length)),
+                    `<name> in 'csv <name> from <url>' is missing`
+                ),
+                createDiagnostic(
+                    Range.create(Position.create(0, 0), Position.create(0, "csv".length)),
+                    `csv has no matching endcsv`
```
Comment (annstriganova) [inline server/src/test/csv.test.ts]: Is it planned to support syntax

```elm
csv <name> from <url>
endcsv
```
?

Comment (PolinaShneider) [reply server/src/test/csv.test.ts]: @VeselovAlex do we need such an option? Currently `csv from url` with closing tag is unsupported.

Comment (VeselovAlex) [reply server/src/test/csv.test.ts]: No, endcsv should not be available here. If csv declaration is completely invalid, we should not require endcsv. In the other hand, endcsv is a good hint to show better warning in future.

**Pull request merged**